File: 016/python/main.py
#
# main.py
# Project Euler - Problem 16
#
# Jordan Sola 2020 - MIT License

import math

# ----- global variables ----- #

base = 2
exponent = 1000

str_result = []

sum = 0

# ---------- #

# math.pow returns a double, needs to be a string
result = str(int(math.pow(base, exponent)))

for i in result:
    str_result.append(i)

# characters in array need to become integers
num_arr = [int(i, base=16) for i in str_result]

for i in num_arr:
    sum += i

print(sum)

File: 016/cpp/src/main.cpp
//
// main.cpp
// Project Euler - Problem 16
//
// Jordan Sola 2020 - MIT License

#include <iostream>
#include <cmath>
#include <string>

using std::cout;
using std::pow;
using std::string;
using std::to_string;

int main()
{
    int base = 2;
    int exponent = 1000;

    double result_doub = pow(base, exponent);
    unsigned long long int result = result_doub;

    string num_to_word = to_string(result);
    size_t word_size = num_to_word.size();

    char num_arr[word_size] = {};

    for (int i = 0; i < word_size; i++)
    {
        num_arr[i] = num_to_word[i];
    }

    unsigned long long sum_of_result = 0;

    for (const int &num : num_arr)
    {
        sum_of_result += num;
    }

    return 0;
}